### DISCUSSION TITLE
How to reset the mAttachCounter variable in MLE

### DISCUSSION THREAD

**Discussion by ArminBreiteneder-eaton** [openthread/openthread#9056] [Q&A] [answered]
In the case I am working on now it can happen that a SED tries to join a network when no active router is present. In this case it will try to attach and on every new attempt it doubles the waiting time. The maximum time is about 20 minutes.

I want to reset the MLE::mAttachCounter variable to restart the attach tries when a user presses a button on the SED to get it into the network faster. I wasn't able to find a possibility to reset mAttachCounter. Is there some other proposed solution to solve that?

Thanks!

**Comment by jwhui** [comment, marked as answer]
Using [otThreadSetEnabled()](https://openthread.io/reference/group/api-thread-general#otthreadsetenabled) to disable then re-enable Thread should achieve what you want.

**Reply by ArminBreiteneder-eaton** [reply]
> Using [otThreadSetEnabled()](https://openthread.io/reference/group/api-thread-general#otthreadsetenabled) to disable then re-enable Thread should achieve what you want.

Ok, that's what I did. I thought there is a more targeted solution. Thank you!
